sample/src/main/java/com/table/sample/MainActivity.kt

```diff
@@ -9,27 +9,47 @@ import android.os.Bundle
 import android.view.View
 import androidx.appcompat.app.AlertDialog
 import androidx.appcompat.app.AppCompatActivity
+import cn.jpush.android.api.JPushInterface
 import co.table.sdk.TableSDK
 import co.table.sdk.android.config.TableLoginCallback
 import co.table.sdk.android.config.UserParams
 import com.google.firebase.messaging.RemoteMessage
+import com.google.gson.Gson
+import com.google.gson.JsonObject
 
 
 class MainActivity : AppCompatActivity(), TableLoginCallback {
     companion object {
         const val EXTRA_REMOTE_MESSAGE = "remote_message"
         const val NOTIFICATION_INTENT_FILTER = "notification_intent_filter"
+        const val JPUSH_NOTIFICATION_INTENT_FILTER = "jpush_notification_intent_filter"
+        const val JPUSH_NOTIFICATION = "jpush_notification"
     }
 
     private var progressDialog: ProgressDialog? = null
-    private var broadcastReceiver: BroadcastReceiver? = null
+    private var firebaseBroadcastReceiver: BroadcastReceiver? = null
+    private var jPushBroadcastReceiver: BroadcastReceiver? = null
+    private var jPushReceiver: JPushReceiver? = null
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
 
+        JPushInterface.setDebugMode(true)
+        JPushInterface.init(this)
+        val regId = JPushInterface.getRegistrationID(this);
+        if (regId != null && regId.isNotEmpty()) {
+            TableSDK.updateJPushRegistrationId(regId)
+        }
+
+        val jpushNotification = intent.getBundleExtra(JPUSH_NOTIFICATION)
+        if (jpushNotification != null) {
+            handleJPushNotification(jpushNotification, this)
+        }
+
```

Comment:
Shall we put some comments around this and the FCM to show which bits are for what in the onCreate()? Pretty obvious in the other files.